### PR TITLE
Kubernetes private docker

### PR DESCRIPTION
**Pull request by xannz**
Added extra configuration options to connect to a private Docker repo using Kubernetes. 
Renamed secret config to password.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -25,4 +25,14 @@ options:
...
       Example usage:
       8080: "expose"
-      8443: ""
\ No newline at end of file
+      8443: ""
+  docker-registry:
+    type: string
+    default: ""
+    description: |
+      Private Docker registry FQDN. Needed when deploying to Kubernetes and image is behind a private repository.
```
Comment (merlijn-sebrechts) [inline config.yaml]: Is this only useful for Kubernetes? If not, leave "Kubernetes" out of the description

File: config.yaml [hunk 2]
```diff
@@ -25,4 +25,14 @@ options:
...
+  docker-email:
+    type: string
+    default: ""
+    description: |
+      Docker email. Needed when deploying to Kubernetes and image is behind a private repository.
```
Comment (merlijn-sebrechts) [inline config.yaml]: Which email is this? Of the Docker user that is used to login?

File: reactive/docker-image.py [hunk 3]
```diff
@@ -54,20 +54,22 @@ def host_connected(dh_relation):
...
         'image': conf.get('image'),
-        'unit': os.environ['JUJU_UNIT_NAME']
+        'unit': os.environ['JUJU_UNIT_NAME'],
+        'docker-registry': conf.get('docker-registry'),
+        'docker-email': conf.get('docker-email')
     }
     log(container_request)
     username = conf.get('username')
-    secret = conf.get('secret')
-    if username != '' or secret != '':
```
Comment (merlijn-sebrechts) [inline reactive/docker-image.py:62]: I think you can just do `if username and secret:`

File: config.yaml (hunk 1 above)
Comment (xannz) [reply config.yaml]: Currently these options are only used by Kubernetes, maybe they will find a use in the layer-image-host.

**Pull request merged**